lib/chef/provider/package/windows.rb

```diff
@@ -80,6 +93,63 @@ def install_package(name, version)
         def remove_package(name, version)
           package_provider.remove_package(name, version)
         end
+
+        # @return [Array] new_version(s) as an array
+        def new_version_array
+          # Because the one in the parent caches things
+          [new_resource.version]
+        end
+
+        private
+
+        def should_download?
+          is_url?(new_resource.source) && !::File.exists?(source_location)
+        end
+
+        def resource_for_provider
+          @resource_for_provider = Chef::Resource::WindowsPackage.new(new_resource.name).tap do |r|
+            r.source(Chef::Util::PathHelper.validate_path(source_location))
+            r.timeout(new_resource.timeout)
+            r.returns(new_resource.returns)
+            r.options(new_resource.options)
+          end
+        end
+
+        def download_source_file
+          source_resource.run_action(:create)
+          Chef::Log.debug("#{@new_resource} fetched source file to #{source_resource.path}")
+        end
+
+        def source_resource
+          @remote_file ||= Chef::Resource::RemoteFile.new(source_location, run_context).tap do |r|
+            r.source(new_resource.source)
+            r.backup(false)
+          end
+        end
+
+        def source_location
+          @source_location ||= begin
+            if is_url?(new_resource.source)
+              uri = ::URI.parse(new_resource.source)
+              filename = ::File.basename(::URI.unescape(uri.path))
+              file_cache_dir = Chef::FileCache.create_cache_path("package/")
+              Chef::Util::PathHelper.cleanpath("#{file_cache_dir}/#{filename}")
+            else
+              Chef::Util::PathHelper.cleanpath(new_resource.source)
+            end
+          end
+        end
+
+        def is_url?(source)
```

Comment:
I've mixed too many concepts here. `uri_scheme?` shouldn't care about what schemes are supported, seems like remote_file's job. I'm going to update it for that, let me know what you think.